### PR TITLE
REACT-477  Logic for handling outside click and bundle improvements 

### PR DESCRIPTION
**Pull request by Andrrii**

### CODE REVIEW

File: src/components/useClickOutside.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import {useEffect, useRef} from 'react';
+import type {UseClickOutsideHandlerProps} from '../types';
+
+export const useClickOutside = ({
+  ref,
+  onOutsideClick,
+  disabled,
+  mouseEvent = 'mousedown',
+  options = true,
+}: UseClickOutsideHandlerProps) => {
```
Comment (slavik-chapelskyi) [inline src/components/useClickOutside.tsx]:
```
export const useClickOutside = (ref, handler, restOptions?) => {
```
lets pass argument like this

Comment (slavik-chapelskyi) [reply src/components/useClickOutside.tsx]: And renameOptions to listenerOptions, so its explicit

File: src/index.test.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+describe('useClickOutside', () => {
+  it('should call onOutsideClick when clicked outside ref element', () => {
+    const ref = {current: document.createElement('div')};
+
+    renderUseClickOutside(ref)
+
+    fireEvent.mouseDown(document);
+
+    expect(onOutsideClick).toHaveBeenCalled();
```
Comment (slavik-chapelskyi) [inline src/index.test.tsx:30]: Lets also cover its get called with event as argumenb

File: src/index.test.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+    fireEvent.mouseDown(innerElement);
+
+    expect(onOutsideClick).not.toHaveBeenCalled();
+  });
+
+  it('should not call onOutsideClick when disabled is true', () => {
+    const ref = {current: document.createElement('div')};
+
+    renderUseClickOutside(ref,true)
```
Comment (slavik-chapelskyi) [inline src/index.test.tsx]:
```suggestion
    renderUseClickOutside(ref, {disabled: true})
```
More explicit & easier to consume without need to look into definition of this function

File: src/index.test.tsx (hunk 2 above)
Comment (Andrrii) [reply src/index.test.tsx:30]: added new test case

**Review by slavik-chapelskyi — APPROVED** [review]
🔥 👍

**Pull request merged**